calypso/service.go

```diff
@@ -434,6 +450,36 @@ func (s *Service) ReshareLTS(req *ReshareLTS) (*ReshareLTSReply, error) {
 	return &ReshareLTSReply{}, nil
 }
 
+// Private service endpoint that extends the set of valid peers with the ones
+// in the provided roster.
+func (s *Service) updateValidPeers(req *updateValidPeers) (
+	*updateValidPeersReply, error) {
+	err := s.verifyProof(&req.Proof)
+	if err != nil {
+		return nil, xerrors.Errorf("verifying proof: %v", err)
+	}
+
+	newRoster, ltsID, err := s.getLtsRoster(&req.Proof)
+	if err != nil {
+		return nil, xerrors.Errorf("retrieving roster: %v", err)
+	}
+
+	s.storage.Lock()
+	currentRoster := s.storage.Rosters[ltsID]
+	s.storage.Unlock()
+
+	// Build the new set as the union of the current and new rosters
```

Comment:
This also stems from my wrong understanding of the calypso re-sharing. We don't need the union here, only the new roster.

Reply:
Changed to only use the new roster